lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -16,12 +16,13 @@ def run(&block)
           queue = build_queue
           lock = Mutex.new
 
-          @workers = thread_count.times.map do
-            Thread.new do
+          @workers = Array.new(thread_count) do
+            t = Thread.new do
               while (item = next_item(queue, lock))
                 yield item
               end
             end
+            t.abort_on_exception = true
```

Comment:
@brynary Since `Thread#value` is meant to be implemented on top of `Thread#join`, I think it would suffer from the same bug where the exception wasn't raised. [This JRuby issue](https://github.com/jruby/jruby/issues/2354) is what's motivating this change.